### PR TITLE
25 react redux jsx

### PR DESCRIPTION
**Pull request by Bellisa**

### CODE REVIEW

File: src/Components/ComUserNav.js [hunk 1]
```diff
@@ -23,7 +22,7 @@ export const ComUserNav = ({
             <NavLink to="/profile" className="dropdown-item" >
               Profile
             </NavLink>
-            <NavLink to="/" className="dropdown-item" onClick={() => { logOut(); setLoginState(null); }} >
+            <NavLink to="/" className="dropdown-item" onClick={() => setLoginState()} >
```
Comment (iliatcymbal) [inline src/Components/ComUserNav.js]: 1) pass to event handler just a function, do not create new function
```
 onClick={setLoginState}
```

2) When I click logout link, browser is reloaded. It shouldn't.

File: src/Components/ComUserTasksInfo.js [hunk 2]
```diff
@@ -1,32 +1,19 @@
...
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (prevState.info !== nextProps.info) {
-      return { info: nextProps.info };
-    }
-    return null;
-  }
 
   render() {
+    const { user } = this.props || {};
+    const { info } = this.props || {};
```
Comment (iliatcymbal) [inline src/Components/ComUserTasksInfo.js:7]: 1) const { user, info } = this.props || {};
2) is there any condition, when props is not exists?

File: src/Pages/TaskList/TaskList.js [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-import { getTodos, deleteTask, updateTask } from '../../services';
+import { UpdateTodoAsync, DeleteTodoByIdAsync, GetAllTodosAsync } from '../../store';
 
-export class TaskList extends Component {
+export class Tasks extends Component {
   constructor(props) {
     super(props);
     let day = this.props.location.search.replace(/\D+/, '') || '';
     if (day === '') {
       day = new Date().getDay();
```
Comment (iliatcymbal) [inline src/Pages/TaskList/TaskList.js]: do you need this?

File: src/app.component.js [hunk 4]
```diff
@@ -6,40 +6,25 @@ import 'jquery';
...
   }
 
-  setLoginState = (data) => {
-    console.log('user', data);
-    this.props.dispatch(SetUser(data));
-  }
   render() {
-    const { user } = this.props;
-
+    const { user } = this.props || {};
```
Comment (iliatcymbal) [inline src/app.component.js:24]: why do you check props?

File: src/parts/Header/Header.js [hunk 5]
```diff
@@ -1,9 +1,22 @@
...
+    const { user } = this.props;
+
+    return (
+      <header className="header">
+        <Navigation pages={pages} user={user} setLoginState={this.setLoginState} />
+      </header>
+    );
+  }
+}
+export const Header = connect(({ user }) => ({ user }))(HeaderComponent);
```
Comment (iliatcymbal) [inline src/parts/Header/Header.js]: I think, Header doesn't require to be connected to the store. But Navigation requires.

File: src/store/reducers.js [hunk 6]
```diff
@@ -17,13 +18,30 @@ export const user = (state = false, { type, data }) => {
...
+export const todos = (state = [], action) => {
+  console.log(action);
 
+  switch (action.type) {
+    case ADD_TASK: {
+      const newState = [...state, action.task];
+      return newState;
+    }
+    case DELETE_TASK_BY_ID: {
+      const newState = state.map(day => day.filter(e => e.id !== action.id).map(ts => ({ ...ts })));
```
Comment (iliatcymbal) [inline src/store/reducers.js]: do not need to make _map(...)_

File: src/store/reducers.js [hunk 7]
```diff
@@ -17,13 +18,30 @@ export const user = (state = false, { type, data }) => {
...
+      return newState;
+    }
+    case UPDATE_TASK: {
+      const newState = state.map(day => day.map(ts => (ts.id === action.task.id ? { ...action.task } : { ...ts })));
+      return newState;
+    }
+    case GET_TASK_BY_ID: {
+      const newState = state.map(day => day.filter(e => e.id === action.id).map(ts => ({ ...ts })));
```
Comment (iliatcymbal) [inline src/store/reducers.js]: do not need to make _map(...)_

File: src/store/sagas/tasksSaga.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+} from '../actions';
+import {
+  getTodos,
+  deleteTask,
+  updateTask
+} from '../../services';
+
+export function* getTodosAsync() {
+  const tasks = yield getTodos()
+    .catch(err => console.log(err, ' get tasks error'));
```
Comment (iliatcymbal) [inline src/store/sagas/tasksSaga.js]: when you're using generators, there is no need to use catch or some other async stuff

File: src/store/sagas/tasksSaga.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    yield put(SetAllTodos(tasks));
+  }
+}
+export function* watchGetTodosAsync() {
+  yield takeEvery(GET_ALL_TASKS_ASYNC, getTodosAsync);
+}
+
+export function* updateTodosAsync(data) {
+  const tasks = yield updateTask(data.task)
+    .catch(err => console.log(err, ' update tasks error'));
```
Comment (iliatcymbal) [inline src/store/sagas/tasksSaga.js]: when you're using generators, there is no need to use catch or some other async stuff

File: src/store/sagas/tasksSaga.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    yield put(UpdateTodo(tasks));
+  }
+}
+export function* watchUpdateTodoAsync() {
+  yield takeEvery(UPDATE_TASK_ASYNC, updateTodosAsync);
+}
+
+export function* deleteTodosAsync(data) {
+  const tasks = yield deleteTask(data.id)
+    .catch(err => console.log(err, ' deletes tasks error'));
```
Comment (iliatcymbal) [inline src/store/sagas/tasksSaga.js]: when you're using generators, there is no need to use catch or some other async stuff

**Review by iliatcymbal — COMMENTED** [review]
Note, that `window.confirm()` good only for development, not for production
```
onActionsClick = (el, act) => {
    if (!confirm(`Are you sure (${act}) '${el.title}'?`)) { return; }
```

User shouldn't see "login" link when logined http://joxi.ru/J2bln9OFXQpdor

Please, remove all _console.log_, you can use it only for yourself.

If you want to name you components with some prefix, use full name, i.e. "Component" instead of "Com"

**Comment by Bellisa** [comment]
@iliatcymbal  Please, review last commit 'implementation sagas.'

File: src/Pages/Pages.js [hunk 11]
```diff
@@ -1,14 +1,11 @@
...
-        <RoutePublic {...this.props} />
-      );
-  }
-}
+export const Pages = props => (props.user ?
+  (
+    <RouteProtected {...props} />
+  ) :
+  (
+    <RoutePublic {...props} />
```
Comment (iliatcymbal) [inline src/Pages/Pages.js]: no need to pass props

File: src/store/actions.js [hunk 12]
```diff
@@ -47,3 +47,7 @@ export const DeleteTodoById = id => ({ type: DELETE_TASK_BY_ID, id });
 
 export const DELETE_TASK_BY_ID_ASYNC = 'Delete task by id async';
 export const DeleteTodoByIdAsync = id => ({ type: DELETE_TASK_BY_ID_ASYNC, id });
+
+// ///////////// ERROR //////////////////////////////////////
```
Comment (iliatcymbal) [inline src/store/actions.js]: if you want to make comments it is better to use something common, like http://usejsdoc.org/

**Review by iliatcymbal — APPROVED** [review]
Looks good

**Pull request merged**